File: CebuNative/Cebu/Core/Data/Models/TranscriptionCost.swift
/**
 * Input: Core Data managed object context
 * Output: TranscriptionCost entity
 * Pos: Core Data model representing API transcription costs
 * If this file is updated, you must update this header and the parent folder's README.md.
 */

import Foundation
import CoreData

@objc(TranscriptionCost)
public class TranscriptionCost: NSManagedObject {
    @NSManaged public var id: UUID?
    @NSManaged public var date: Date?
    @NSManaged public var duration: Double
    @NSManaged public var cost: Double
    @NSManaged public var provider: String?
    @NSManaged public var entryId: UUID?
}

extension TranscriptionCost {
    @nonobjc public class func fetchRequest() -> NSFetchRequest<TranscriptionCost> {
        return NSFetchRequest<TranscriptionCost>(entityName: "TranscriptionCost")
    }
}

File: cebu-expo/src/core/data/models/TranscriptionCost.ts
/**
 * Input: WatermelonDB Model class
 * Output: TranscriptionCost model for tracking API costs
 * Pos: Cost tracking entity for transcription API usage
 * If this file is updated, you must update this header and the parent folder's README.md.
 */

import { Model } from '@nozbe/watermelondb';
import { field, readonly, date } from '@nozbe/watermelondb/decorators';

export default class TranscriptionCost extends Model {
  static table = 'transcription_costs';

  @date('date') date!: Date;
  @field('duration') duration!: number;
  @field('cost') cost!: number;
  @field('provider') provider!: string;
  @field('entry_id') entryId?: string;
}
